# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/webauthn/WebAuthnResult.h
# Commit: 618505b0336f
# Full Hash: 618505b0336f811d1f53201a25151a5d440957dd
# Author: John Schanck <jschanck@mozilla.com>
# Date: 2024-12-07 09:10:49
# Description:
#   Bug 1935713 - check webauthn.dll version before accessing WebAuthn PRF assertion values. r=dveditz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231431
# ==============================================================================

diff -r 596701f589a2 -r 618505b0336f dom/webauthn/WebAuthnResult.h
--- a/dom/webauthn/WebAuthnResult.h	Fri Dec 06 18:57:51 2024 +0000
+++ b/dom/webauthn/WebAuthnResult.h	Fri Dec 06 18:59:29 2024 +0000
@@ -218,16 +218,18 @@
                                       aResponse->cbAuthenticatorData);
 
     mAuthenticatorAttachment = Nothing();  // not available
-    if (aResponse->pHmacSecret) {
-      if (aResponse->pHmacSecret->cbFirst > 0) {
-        mPrfFirst.emplace();
-        mPrfFirst->AppendElements(aResponse->pHmacSecret->pbFirst,
-                                  aResponse->pHmacSecret->cbFirst);
-      }
-      if (aResponse->pHmacSecret->cbSecond > 0) {
-        mPrfSecond.emplace();
-        mPrfSecond->AppendElements(aResponse->pHmacSecret->pbSecond,
-                                   aResponse->pHmacSecret->cbSecond);
+    if (aResponse->dwVersion >= WEBAUTHN_ASSERTION_VERSION_3) {
+      if (aResponse->pHmacSecret) {
+        if (aResponse->pHmacSecret->cbFirst > 0) {
+          mPrfFirst.emplace();
+          mPrfFirst->AppendElements(aResponse->pHmacSecret->pbFirst,
+                                    aResponse->pHmacSecret->cbFirst);
+        }
+        if (aResponse->pHmacSecret->cbSecond > 0) {
+          mPrfSecond.emplace();
+          mPrfSecond->AppendElements(aResponse->pHmacSecret->pbSecond,
+                                     aResponse->pHmacSecret->cbSecond);
+        }
       }
     }
   }
